### PR TITLE
[UI] As a user, I can see the home page with the filtered message list

### LINKED ISSUE #20: [UI] As a user, I can see the home page with the filtered message list
## Why

User can see the filtered message

## Acceptance Criteria

- Show "All Filters"
- Show Add filter button

### PR DESCRIPTION
**Pull request by lydiasama**
## What happened 👀

UI for home screen
 
## Insight 📝

- The action of the `Add filter button` will add later.
- The `Button` we need to use `android.widget.Button` because we use `Theme.MaterialComponents.NoActionBar` as the app theme.
Ref: https://stackoverflow.com/a/64839917

### CODE REVIEW

**Comment by hoangnguyen92dn** [comment]
@lydiasama I rebased #6. Please rebase this PR then we will start reviewing 😄

**Comment by lydiasama** [comment]
@hoangnguyen92dn This PR is ready to review. 🙏🏻

**Comment by lydiasama** [comment]
Will add more change for the item of RecyclerView 🙏🏻

File: android/app/src/main/res/drawable/bg_circle_button.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval">
+
+    <solid android:color="@color/purple_violent_violet" />
+
+    <corners android:radius="20dp" />
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/drawable/bg_circle_button.xml]: OPTIONAL: Extract 20dp to dimension for default radius.

File: android/app/src/main/res/drawable/bg_home.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <corners android:radius="20dp" />
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/drawable/bg_home.xml]: Same here

File: android/app/src/main/res/layout/fragment_home.xml [hunk 3]
```diff
@@ -3,24 +3,40 @@
...
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:background="@drawable/bg_app"
+    android:padding="@dimen/default_margin_16">
+
+    <View
+        android:id="@+id/vHomeBackground"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/layout/fragment_home.xml]: Why don't we set match_parent and get rid of constraint top/bottom?

File: android/app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -3,24 +3,40 @@
...
+        android:background="@drawable/bg_home"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
     <androidx.recyclerview.widget.RecyclerView
-        android:id="@+id/rvHomeData"
+        android:id="@+id/rvHomeSmsData"
         android:layout_width="match_parent"
         android:layout_height="0dp"
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/layout/fragment_home.xml]: Same here

File: android/app/src/main/res/layout/item_data.xml [hunk 5]
```diff
@@ -4,43 +4,44 @@
...
 
-    <androidx.appcompat.widget.AppCompatImageView
-        android:id="@+id/ivDataThumbnail"
-        android:layout_width="100dp"
-        android:layout_height="0dp"
-        android:scaleType="centerCrop"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintHeight_min="100dp"
+    <ImageView
+        android:id="@+id/ivSmsBullet"
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/layout/item_data.xml]: Should we rename `item_data` to `item_sms` for consistency?
- `item_data` -> `ivData`
- `item_sms` -> `ivSms`

Comment (hoangnguyen92dn) [reply android/app/src/main/res/layout/item_data.xml]: Btw, I think we should add the label for the datetime.

File: android/app/src/main/res/layout/fragment_home.xml [hunk 6]
```diff
@@ -3,24 +3,40 @@
...
+        android:id="@+id/vHomeBackground"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:background="@drawable/bg_home"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
     <androidx.recyclerview.widget.RecyclerView
-        android:id="@+id/rvHomeData"
+        android:id="@+id/rvHomeSmsData"
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/layout/fragment_home.xml:17]: Should we add the empty page if the list is empty? or at least add the TODO note 😄

File: android/app/src/main/res/layout/fragment_home.xml (hunk 3 above)
Comment (lydiasama) [reply android/app/src/main/res/layout/fragment_home.xml]: Oh, it can. Thanks. 🙏🏻

File: android/app/src/main/res/layout/item_data.xml (hunk 5 above)
Comment (lydiasama) [reply android/app/src/main/res/layout/item_data.xml]: Let's do it later since the endpoint doesn't include this field for now. 🙏🏻

**Comment by lydiasama** [comment]
@hoangnguyen92dn  All updated. 🙏🏻

**Review by hoangnguyen92dn — APPROVED** [review]
LGTM. Test classes will be updated on #9

**Pull request merged**